src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/MsQuicStream.cs

```diff
@@ -383,6 +412,14 @@ internal override async ValueTask ShutdownWriteCompleted(CancellationToken cance
         {
             ThrowIfDisposed();
 
+            lock (_state)
```

Comment:
I'm. not sure if this needs to be under lock since we are not modifying the state. 
For the check there is IMHO race condition anyway e.g. the state can be changed immediately after the lock before other code executes.

Reply:
You technically still need the lock to make visible changes from other threads. We could make access to that one field use `Volatile.Read`/`Volatile.Write` as an alternative to avoid the lock.

Reply:
yes. but even with that there is still race to this really becomes best effort IMHO. 
I'm Im not sure if this makes any measurable impact so I feel it is ok to keep it for now.